testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpClientTest.java

```diff
@@ -1043,6 +1066,47 @@ SpanDataAssert assertClientSpan(
             });
   }
 
+  @SuppressWarnings("unchecked")
+  private static <T> AttributeKey<T> getAttributeKey(AttributeKey<T> oldKey) {
```

Comment:
Originally I thought of adding separate `if`s in the assertion method for the new attributes, but this is even better 👍